app/src/main-process/main.ts

```diff
@@ -43,100 +36,49 @@ const launchTime = now()
 
 let preventQuit = false
 let readyTime: number | null = null
-let hasReportedUncaughtException = false
 
 type OnDidLoadFn = (window: AppWindow) => void
 /** See the `onDidLoad` function. */
 let onDidLoadFns: Array<OnDidLoadFn> | null = []
 
-function uncaughtException(error: Error) {
-  log.error(formatError(error))
-
-  if (hasReportedUncaughtException) {
-    return
-  }
-
-  hasReportedUncaughtException = true
+function handleUncaughtException(error: Error) {
   preventQuit = true
 
-  setCrashMenu()
-
-  const isLaunchError = !mainWindow
-
   if (mainWindow) {
     mainWindow.destroy()
     mainWindow = null
   }
 
-  const crashWindow = new CrashWindow(
-    isLaunchError ? 'launch' : 'generic',
-    error
-  )
-
-  crashWindow.onDidLoad(() => {
-    crashWindow.show()
-  })
-
-  crashWindow.onFailedToLoad(() => {
-    dialog.showMessageBox(
-      {
-        type: 'error',
-        title: __DARWIN__ ? `Unrecoverable Error` : 'Unrecoverable error',
-        message:
-          `GitHub Desktop has encountered an unrecoverable error and will need to restart.\n\n` +
-          `This has been reported to the team, but if you encounter this repeatedly please report ` +
-          `this issue to the GitHub Desktop issue tracker.\n\n${error.stack ||
-            error.message}`,
-      },
-      response => {
-        if (!__DEV__) {
-          app.relaunch()
-        }
-        app.quit()
-      }
-    )
-  })
-
-  crashWindow.onClose(() => {
-    if (!__DEV__) {
-      app.relaunch()
-    }
-    app.quit()
-  })
-
-  crashWindow.load()
+  const isLaunchError = !mainWindow
+  showUncaughtException(isLaunchError, error)
 }
 
 process.on('uncaughtException', (error: Error) => {
   error = withSourceMappedStack(error)
 
   reportError(error)
-  uncaughtException(error)
+  handleUncaughtException(error)
 })
 
-let willQuit = false
-
+let handlingSquirrelEvent = false
 if (__WIN32__ && process.argv.length > 1) {
   const arg = process.argv[1]
+
   const promise = handleSquirrelEvent(arg)
   if (promise) {
-    willQuit = true
+    handlingSquirrelEvent = true
     promise
-      .then(() => {
-        app.quit()
-      })
       .catch(e => {
         log.error(`Failed handling Squirrel event: ${arg}`, e)
       })
+      .then(() => {
```

Comment:
Yeah, with the `catch` before the `then` it ensures we always quit. The other way it wouldn't.